omegaconf/_utils.py

```diff
@@ -351,11 +368,11 @@ def get_value_kind(
         this parsing step is skipped: this is more efficient, but will not detect errors.
     """
 
-    value = _get_value(value)
-
-    if isinstance(value, str) and value == "???":
+    if _is_missing_value(value):
         return ValueKind.MANDATORY_MISSING
 
+    value = _get_value(value)
```

Comment:
What about keeping `value = _get_value(value)` first and calling `_is_missing_literal()` on it? (seems a bit more efficient, though it may not matter in practice)

Reply:
_get_value is actually only unwrapping value nodes so it's not quiet the same as what we are after here.

Reply:
True, there's a subtle difference, but from what I can see it doesn't matter here (in the sense that both approaches should behave the same), and my suggestion actually better matches the previous implementation, right?

Reply:
you are welcome to try to change it and see the test failure.

Reply:
Also related to the assert => resolving